### ISSUE TITLE
Human Readable File Names

### ISSUE THREAD

**Issue by Shrinks99** [dularion/streama#393]
While Streama does an excellent job at providing a nice front end for media management the organization of the media folder provides a less-than-optimal experience.  It is hard to figure out what "ece2f3e2c3375ab65f54d7796fbf4bedbdea0bf936564c4f8cb30d146adf6c24.mp4" actually is just by looking at the file name.  Would it be possible to store these as "SHOWNAME-S01E01-1080p-2.5GB.mp4" (or a similar format) instead?  This would be helpful for file management.

**Comment by PiperMp3** [comment]
Well, I think its how Streama indexes the files you add, sort of creates a internal link between the file and the database somehow. It's probably possible to do it differently, but I think thats the reason for it.

Just so Streama (system) knows what the heck file is for what series, and season and episode, even after a restart.

Think it's easier to have Streama create it's own filenames, instead of forcing Streama to understand yours - something along those lines perhaps.
I get your point though and it's probably something Streama may get in the future maybe, like an enhancement.

**Comment by Shrinks99** [comment]
> I think its how Streama indexes the files you add, sort of creates a internal link between the file and the database

Yeah, that's exactly how it works and it's definitely easier.  The only issue with it is that it results in poor file organization if somebody wanted to transfer their media elsewhere.

**Comment by PiperMp3** [comment]
Yea of course, I fully see that, and agree - hopefully a future enhancement!

**Comment by dularion** [comment]
This is truly something I've noticed too, and personally using the local file explorer helped me circumvent this issue: I just rsync my files into the designated 'local files' directory and then pick out the files from there rather than upload them via streama. That way I can organize stuff into folders, too. The problem comes with duplicate names etc. If you were to attempt to upload two .srt files with the same name then the application wouldn't be able to upload the second because it doesn't know whether it's allowed to replace it or what. So then I'd either have to implement a non-duplication handling where files are suffixed with a counter, but that's also quite complex, or I just override the file. But what you see in the upload directory currently is a lot more solid than a file-name. It's a hash over the file content itself. Basically the string represents the file itself. So if you upload a file called 'episode 1.mp4' and there's already such a file for a completely different show then the hash for both files would be entirely different whereas the name wouldn't be. You know?

**Comment by Shrinks99** [comment]
I figured it was done by file hashes.  If the program automatically renames things (and maybe also puts TV shows in to folders with seasons also organized by folders) duplicate file names shouldn't be a huge issue.  When content is uploaded check for duplicate items and add a 1 if there is a duplicate?  Using rsync and choosing the file rather than uploading it through the web interface makes a lot of sense with the current system but it's not the most user friendly and all.

**Comment by kunaltyagi** [comment]
How about using `<name>:sXXX:eXXX:rXXX:<hash>.<ext>` or a similar super-set? You'd need to find the last location of the delimiter (here `:`) and everything after that to `.` is the unique content based hash. (s is season, e is episode, r is resolution, etc.)
Another option is organizing name, season, episode and keeping the files hashed. So the srt and media file are together in a folder hierarchy of `<name>/sXXX/eXXX/rXXX`. Does this count as human readable: No; human understandable: Yes
